File: Colaveco/JavaModel/mobile/iOS/gxcommon/downloadpdf.swift
/*
               File: DownloadPDF
        Description: Download PDF
             Author: GeneXus Swift Generator version 17_0_7-156546
       Generated on: November 14, 2022 10:7:19.64
       Program type: Callable routine
          Main DBMS: SQLite
*/

@objc( downloadpdf )
public final  class downloadpdf: GXProcedure
{
   @objc public func executeFlexibleClient( _ aP0: NSNumber? ,
                                            _ aP1: NSString? )
   {
      let iP0: Int64 = (aP0 == nil) ? 0 : aP0!.int64Value as Int64
      let iP1: String = (aP1 == nil) ? "" : aP1! as String
      execute(iP0, iP1)
   }

   public func execute( _ aP0: Int64 ,
                        _ aP1: String )
   {
      execute_int(aP0, aP1)
   }

   private func execute_int( _ aP0: Int64 ,
                             _ aP1: String )
   {
      self.AV10resultadoFicha = aP0
      self.AV9resultadoPathPdf = aP1
      initialize()
      /* GeneXus formulas */
      /* Output device settings */
      autoreleasepool
      {
         privateExecute()
      }
   }

   private func privateExecute( )
   {
      AV8url = "https://www.colaveco.com.uy/gestor/utils/bajar_archivo.php?archivo="  +  AV9resultadoPathPdf
      loger().execute( AV8url)
      AV12path = GXDirectory.applicationdatapath()  +  AV11file.separator  +  String.gxStringFromNumber(Decimal(AV10resultadoFicha), totalLength:18, decimals:0).gxStringByTrimming()  +  ".pdf"
      loger().execute( AV12path)
      AV11file.source = AV12path
      AV15Uri = "file://"  +  AV12path
      loger().execute( AV15Uri)
      AV16Destfile = AV15Uri
      cleanup()
   }

   public override func cleanup( )
   {
      closeOpenCursors()
   }

   private func closeOpenCursors( )
   {
   }

   /* Aggregate/select formulas */
   public override func initialize( )
   {
      AV8url = ""
      AV12path = ""
      AV11file = GXFile()
      AV15Uri = ""
      AV16Destfile = ""
      /* GeneXus formulas. */
      Gx_err = 0
   }

   private var Gx_err : Int = 0
   private var AV10resultadoFicha : Int64 = 0
   private var AV15Uri : String = ""
   private var AV16Destfile : String = ""
   private var AV9resultadoPathPdf : String = ""
   private var AV8url : String = ""
   private var AV12path : String = ""
   private var AV11file : GXFile = GXFile()
}

File: Colaveco/JavaModel/mobile/Android/gxcommon/com/colaveco/downloadpdf.java
/*
               File: DownloadPDF
        Description: Download PDF
             Author: GeneXus Android Generator version 17_0_7-156546
       Generated on: November 14, 2022 10:7:19.25
       Program type: Callable routine
          Main DBMS: SQLite
*/
package com.colaveco ;
import com.colaveco.*;
import com.genexus.*;
import com.genexus.db.*;
import java.sql.*;
import com.artech.base.services.*;

public final  class downloadpdf extends GXProcedure implements IGxProcedure
{
   public boolean execute( IPropertiesObject androidPropertiesObject )
   {

      long aP0 = 0;
      String aP1 = "";

      aP0 = (long) GXutil.lval( androidPropertiesObject.optStringProperty("resultadoFicha"));
      aP1 = (String) androidPropertiesObject.optStringProperty("resultadoPathPdf");

      execute(aP0, aP1);



      return true ;
   }

   public downloadpdf( int remoteHandle )
   {
      super( remoteHandle , new ModelContext( downloadpdf.class ), "" );
   }

   public downloadpdf( int remoteHandle ,
                       ModelContext context )
   {
      super( remoteHandle , context, "" );
   }

   public void execute( long aP0 ,
                        String aP1 )
   {
      execute_int(aP0, aP1);
   }

   private void execute_int( long aP0 ,
                             String aP1 )
   {
      downloadpdf.this.AV10resultadoFicha = aP0;
      downloadpdf.this.AV9resultadoPathPdf = aP1;
      initialize();
      /* GeneXus formulas */
      /* Output device settings */
      privateExecute();
   }

   private void privateExecute( )
   {
      AV8url = "https://www.colaveco.com.uy/gestor/utils/bajar_archivo.php?archivo=" + AV9resultadoPathPdf ;
      new com.colaveco.loger(remoteHandle, context).execute( AV8url) ;
      AV12path = com.genexus.util.GXDirectory.applicationdatapath() + AV11file.getSeparator() + GXutil.trim( GXutil.str( AV10resultadoFicha, 18, 0)) + ".pdf" ;
      new com.colaveco.loger(remoteHandle, context).execute( AV12path) ;
      AV11file.setSource( AV12path );
      AV15Uri = "file://" + AV12path ;
      new com.colaveco.loger(remoteHandle, context).execute( AV15Uri) ;
      AV16Destfile = AV15Uri ;
      cleanup();
   }

   protected void cleanup( )
   {
      CloseOpenCursors();
      exitApplication();
   }

   protected void CloseOpenCursors( )
   {
   }

   /* Aggregate/select formulas */
   public void initialize( )
   {
      AV8url = "" ;
      AV12path = "" ;
      AV11file = new com.genexus.util.GXFile();
      AV15Uri = "" ;
      AV16Destfile = "" ;
      /* GeneXus formulas. */
      Gx_err = (short)(0) ;
   }

   private short Gx_err ;
   private long AV10resultadoFicha ;
   private String AV15Uri ;
   private String AV16Destfile ;
   private String AV9resultadoPathPdf ;
   private String AV8url ;
   private String AV12path ;
   private com.genexus.util.GXFile AV11file ;
}

